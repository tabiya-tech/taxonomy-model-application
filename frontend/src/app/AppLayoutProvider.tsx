import React, { useState, createContext, useContext } from 'react';
import {AppLayoutProviderProps, AppLayoutStateType} from './app.type'

export const AppLayoutContext = createContext<
  AppLayoutStateType | undefined
>(undefined);


export const useAppLayout = () => {
  const context = useContext(AppLayoutContext);
  if (!context) {
    throw new Error(
      'useAppLayout must be used within an AppLayoutProvider'
    );
  }
  return context;
};


export const AppLayoutProvider: React.FC<
  AppLayoutProviderProps
> = ({ children }) => {
  const [contentHeader, setContentHeader] =
    useState<React.ReactNode>();

  return (
    <AppLayoutContext.Provider
      value={{ contentHeader, setContentHeader }}
    >
      {children}
    </AppLayoutContext.Provider>
  );
};

export default AppLayoutProvider;