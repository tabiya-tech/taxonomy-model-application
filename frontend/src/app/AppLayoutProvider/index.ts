export * from './AppLayoutProvider'
export * from './AppLayoutContext'