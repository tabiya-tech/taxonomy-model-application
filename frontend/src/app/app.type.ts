import React from 'react';

export interface AppLayoutStateType {
  contentHeader?: React.ReactNode;
  setContentHeader: React.Dispatch<
    React.SetStateAction<React.ReactNode>
  >;
}

export interface AppLayoutProviderProps {
  children: React.ReactNode;
}
