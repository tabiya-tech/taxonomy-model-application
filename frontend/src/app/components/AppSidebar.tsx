import { NavLink } from 'react-router-dom';
import Box from '@mui/material/Box';
import FolderOutlinedIcon from '@mui/icons-material/FolderOutlined';
import SettingsIcon from '@mui/icons-material/Settings';
import PermIdentityIcon from '@mui/icons-material/PermIdentity';
import EditOutlinedIcon from '@mui/icons-material/EditOutlined';
import VerticalSplitOutlinedIcon from '@mui/icons-material/VerticalSplitOutlined';
import { styled } from '@mui/material';

const uniqueId = '579896ed-8a99-4a1e-8dc4-e4236d7b06df';
export const DATA_TEST_ID = {
  APP_SIDEBAR_CONTAINER: `app-sidebar-container-${uniqueId}`,
  APP_SIDEBAR_ICON_DIRECTORY: `app-sidebar-icon-directory-${uniqueId}`,
  APP_SIDEBAR_ICON_EXPLORE: `app-sidebar-icon-explore-${uniqueId}`,
  APP_SIDEBAR_ICON_EDIT: `app-sidebar-icon-edit-${uniqueId}`,
  APP_SIDEBAR_ICON_USER: `app-sidebar-icon-user-${uniqueId}`,
  APP_SIDEBAR_ICON_SETTINGS: `app-sidebar-icon-settings-${uniqueId}`,
  APP_SIDEBAR_DIRECTORY_LINK: `app-sidebar-directory-link-${uniqueId}`,
  APP_SIDEBAR_EXPLORE_LINK: `app-sidebar-explore-link-${uniqueId}`,
  APP_SIDEBAR_EDIT_LINK: `app-sidebar-edit-link-${uniqueId}`,
  APP_SIDEBAR_USER_LINK: `app-sidebar-user-link-${uniqueId}`,
  APP_SIDEBAR_SETTINGS_LINK: `app-sidebar-settings-link-${uniqueId}`,
};

export const ITEMS_LABEL_TEXT = {
  [DATA_TEST_ID.APP_SIDEBAR_ICON_DIRECTORY]: 'Directory',
  [DATA_TEST_ID.APP_SIDEBAR_ICON_EXPLORE]: 'Explore',
  [DATA_TEST_ID.APP_SIDEBAR_ICON_EDIT]: 'Model',
  [DATA_TEST_ID.APP_SIDEBAR_ICON_USER]: 'Users',
  [DATA_TEST_ID.APP_SIDEBAR_ICON_SETTINGS]: 'Settings',
};

const appSidebarItems = [
  {
    icon: (
      <FolderOutlinedIcon
        sx={{ width: '24px', height: '24px' }}
        data-testid={DATA_TEST_ID.APP_SIDEBAR_ICON_DIRECTORY}
        className='item-icon'
      />
    ),
    label: ITEMS_LABEL_TEXT[DATA_TEST_ID.APP_SIDEBAR_ICON_DIRECTORY],
    pathName: '/',
    iconLinkTestID: DATA_TEST_ID.APP_SIDEBAR_DIRECTORY_LINK,
  },
  {
    icon: (
      <VerticalSplitOutlinedIcon
        sx={{ width: '24px', height: '24px' }}
        data-testid={DATA_TEST_ID.APP_SIDEBAR_ICON_EXPLORE}
        className='item-icon'
      />
    ),
    label: ITEMS_LABEL_TEXT[DATA_TEST_ID.APP_SIDEBAR_ICON_EXPLORE],
    pathName: '/explore',
    iconLinkTestID: DATA_TEST_ID.APP_SIDEBAR_EXPLORE_LINK,
  },
  {
    icon: (
      <EditOutlinedIcon
        sx={{ width: '24px', height: '24px' }}
        data-testid={DATA_TEST_ID.APP_SIDEBAR_ICON_EDIT}
        className='item-icon'
      />
    ),
    label: ITEMS_LABEL_TEXT[DATA_TEST_ID.APP_SIDEBAR_ICON_EDIT],
    pathName: 'edit',
    iconLinkTestID: DATA_TEST_ID.APP_SIDEBAR_EDIT_LINK,
  },
  {
    icon: (
      <PermIdentityIcon
        sx={{ width: '24px', height: '24px' }}
        data-testid={DATA_TEST_ID.APP_SIDEBAR_ICON_USER}
        className='item-icon'
      />
    ),
    label: ITEMS_LABEL_TEXT[DATA_TEST_ID.APP_SIDEBAR_ICON_USER],
    pathName: '/users',
    iconLinkTestID: DATA_TEST_ID.APP_SIDEBAR_USER_LINK,
  },
  {
    icon: (
      <SettingsIcon
        sx={{ width: '24px', height: '24px' }}
        data-testid={DATA_TEST_ID.APP_SIDEBAR_ICON_SETTINGS}
        className='item-icon'
      />
    ),
    label: ITEMS_LABEL_TEXT[DATA_TEST_ID.APP_SIDEBAR_ICON_SETTINGS],
    pathName: '/settings',
    iconLinkTestID: DATA_TEST_ID.APP_SIDEBAR_SETTINGS_LINK,
  },
];

const CustomNavLink = styled(NavLink)`
  text-decoration: none;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 4px;
  color: #43474e;
  .item-icon {
    display: flex;
    align-items: center;
    padding: 4px 14px;
    border-radius: 16px;
  }
  &.active {
    .item-icon {
      background-color: #5cff9f;
    }
  }
  .item-label {
    font-size: 12px;
    color: #43474e;
  }
  &:hover {
    .item-icon {
      background-color: #5cff9f;
    }
  }
`;

const Container = styled(Box)`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 0 16px;
  //margin-top: 44px;
  font-family: ${({ theme }) => theme.typography.fontFamily};
  height: 100%;
  padding-left: 24px;
`;

const AppSidebar = () => {
  return (
    <Container
      gap='12px'
      data-testid={DATA_TEST_ID.APP_SIDEBAR_CONTAINER}
    >
      {appSidebarItems.map((item) => (
        <CustomNavLink key={item.pathName} to={item.pathName} data-testid={item.iconLinkTestID}>
          {item.icon}
          <span className='item-label'>{item.label}</span>
        </CustomNavLink>
      ))}
    </Container>
  );
};
export default AppSidebar;
