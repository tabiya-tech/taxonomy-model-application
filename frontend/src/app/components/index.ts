export * from './AppHeader'
export * from './AppLayout'
export * from './AppSidebar'
export * from './ContentHeaderContainer'