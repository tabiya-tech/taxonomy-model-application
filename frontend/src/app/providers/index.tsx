export * from "./IsOnlineProvider";
export * from "./AuthProvider";
