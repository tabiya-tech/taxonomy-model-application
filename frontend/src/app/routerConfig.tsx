import Info from '../info/Info';
import ModelDirectory from '../modeldirectory/ModelDirectory';

export const routerPaths = {
  ROOT: '/',
  INFO: '/info',
  MODEL_DIRECTORY: '/modeldirectory',
};
const uniqueId = '37d307ae-4f1e-4d8d-bafe-fd642f8af4dc';
export const DATA_TEST_ID = {
  LANDING_PAGE: `landing-page-root${uniqueId}`,
};

export const routerConfig = [
  {
    path: routerPaths.ROOT,
    element: (
      <div data-testid={DATA_TEST_ID.LANDING_PAGE}>
        Welcome Page
      </div>
    ),
    errorElement: <div>Sorry something went wrong</div>,
  },
  {
    path: routerPaths.INFO,
    element: <Info />,
    errorElement: <div>Sorry info got wrong</div>,
  },
  {
    path: routerPaths.MODEL_DIRECTORY,
    element: <ModelDirectory />,
    errorElement: <div>Sorry upload got wrong</div>,
  },
];

export default routerConfig;
