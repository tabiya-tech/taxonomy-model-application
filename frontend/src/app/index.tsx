import { HashRouter, Route, Routes } from "react-router-dom";
import routerConfig from "./routerConfig";
import { AppLayout } from "./components";

export const TaxonomyModelApp = () => {
  return (
    <HashRouter>
      <AppLayout>
        <Routes>
          {routerConfig.map((route) => (
            <Route key={route.path} path={route.path} element={route.element} errorElement={route.errorElement} />
          ))}
        </Routes>
      </AppLayout>
    </HashRouter>
  );
};

export default TaxonomyModelApp;
