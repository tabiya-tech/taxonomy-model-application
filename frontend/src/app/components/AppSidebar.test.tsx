import AppSidebar, {DATA_TEST_ID, ITEMS_LABEL_TEXT} from "./AppSidebar";
import {HashRouter} from "react-router-dom";
import {render, screen, within} from "src/_test_utilities/test-utils";
import {routerPaths} from "src/app/routerConfig";
import {testNavigateToPath} from "src/_test_utilities/routeNavigation";

describe("AppSidebar render tests", () => {

  test("renders correctly", () => {
    // GIVEN a AppSidebar component that has access to the router
    const givenAppSidebar = (
      <HashRouter>
        <AppSidebar/>
      </HashRouter>
    );

    // WHEN it is rendered
    render(givenAppSidebar);

    // THEN expect it to be shown
    const appSidebarContainer = screen.getByTestId(DATA_TEST_ID.CONTAINER);
    expect(appSidebarContainer).toBeInTheDocument();
  });

  const allIconsCases = [
    ["Directory", DATA_TEST_ID.DIRECTORY_LINK, DATA_TEST_ID.DIRECTORY_ICON, ITEMS_LABEL_TEXT.DIRECTORY],
    ["Explore", DATA_TEST_ID.EXPLORE_LINK, DATA_TEST_ID.EXPLORE_ICON, ITEMS_LABEL_TEXT.EXPLORE],
    ["User", DATA_TEST_ID.USER_LINK, DATA_TEST_ID.USER_ICON, ITEMS_LABEL_TEXT.USER],
    ["Settings", DATA_TEST_ID.SETTINGS_LINK, DATA_TEST_ID.SETTINGS_ICON, ITEMS_LABEL_TEXT.SETTINGS],
  ];

  it.each(allIconsCases)
  ("should render %s link correct", (description, linkDataTestId, iconDataTestId, expectedLinkText) => {
      // GIVEN a AppSidebar component that has access to the router
      const givenAppSidebar = (
        <HashRouter>
          <AppSidebar/>
        </HashRouter>
      );

      // WHEN it is rendered
      render(givenAppSidebar);

      // THEN expect the link to be shown
      const linkElement = screen.getByTestId(linkDataTestId);
      expect(linkElement).toBeInTheDocument();

      // AND the icon to be shown wrapped in the link
      const iconElement = within(linkElement).getByTestId(iconDataTestId);
      expect(iconElement).toBeInTheDocument();

      // AND expect the link to have the correct label
      expect(linkElement).toHaveTextContent(expectedLinkText);
    },
  );
});

describe("AppSidebar action tests", () => {
  const allRouterCases = [
    ["Directory", DATA_TEST_ID.DIRECTORY_LINK, routerPaths.MODEL_DIRECTORY],
    ["Explore", DATA_TEST_ID.EXPLORE_LINK, routerPaths.EXPLORE],
    ["Edit", DATA_TEST_ID.EDIT_LINK, routerPaths.EDIT],
    ["User", DATA_TEST_ID.USER_LINK, routerPaths.USERS],
    ["Settings", DATA_TEST_ID.SETTINGS_LINK, routerPaths.SETTINGS],
  ];
  allRouterCases.forEach(([linkName, linkTestId, expectedPath]) => {
    testNavigateToPath(<AppSidebar/>, linkName, linkTestId, expectedPath);
  });
});