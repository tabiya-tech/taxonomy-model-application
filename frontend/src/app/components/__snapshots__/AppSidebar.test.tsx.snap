// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSidebar render tests renders correctly 1`] = `
<div
  class="MuiBox-root css-8aaikx"
  data-testid="app-sidebar-container-579896ed-8a99-4a1e-8dc4-e4236d7b06df"
>
  <a
    class="css-5mjdi8"
    data-testid="app-sidebar-directory-link-579896ed-8a99-4a1e-8dc4-e4236d7b06df"
    href="#/modeldirectory"
  >
    <div
      class="MuiBox-root css-1n6ok9e"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-etbowg-MuiSvgIcon-root"
        data-testid="app-sidebar-directory-icon-579896ed-8a99-4a1e-8dc4-e4236d7b06df"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m9.17 6 2 2H20v10H4V6h5.17M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
        />
      </svg>
    </div>
    <span
      class="MuiTypography-root MuiTypography-caption css-1vghao7-MuiTypography-root"
    >
      Directory
    </span>
  </a>
  <a
    class="css-5mjdi8"
    data-testid="app-sidebar-settings-link-579896ed-8a99-4a1e-8dc4-e4236d7b06df"
    href="#/settings"
  >
    <div
      class="MuiBox-root css-1n6ok9e"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-etbowg-MuiSvgIcon-root"
        data-testid="app-sidebar-settings-icon-579896ed-8a99-4a1e-8dc4-e4236d7b06df"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
        />
      </svg>
    </div>
    <span
      class="MuiTypography-root MuiTypography-caption css-1vghao7-MuiTypography-root"
    >
      Settings
    </span>
  </a>
</div>
`;
